File: tests/fixtures/go/slop_restate_pt_br.go
package main

type ContadorAcessos struct {
	contador int
}

func (c *ContadorAcessos) RegistrarHit() {
	// incrementa o contador
	c.contador += 1
}

func (c *ContadorAcessos) RegistrarLote(n int) {
	c.contador += n // incrementa o contador
}

func main() {
	c := &ContadorAcessos{}
	c.RegistrarHit()
	c.RegistrarLote(3)
}

// expect-line: 8 SLOP042
// expect-line: 13 SLOP042

File: tests/fixtures/rust/slop_restate_pt_br.rs
fn registrar_hit(contador: &mut i32) {
    // incrementa o contador
    *contador += 1;
}

fn registrar_lote(contador: &mut i32, n: i32) {
    *contador += n; // incrementa o contador
}

fn main() {
    let mut acessos = 0;
    registrar_hit(&mut acessos);
    registrar_lote(&mut acessos, 3);
}

// expect-line: 2 SLOP042
// expect-line: 7 SLOP042
